src/bootman/kernel.c

```diff
@@ -608,6 +609,40 @@ static bool boot_manager_remove_legacy_uefi_kernel(const BootManager *manager, c
         return ret;
 }
 
+bool _copy_extra_initrds(const char *initrd_base, const char *target_dir) {
+        char *initrd_source = NULL;
+        autofree(char) *initrd_glob = string_printf("%s.*", initrd_base);
+        autofree(char) *initrd_target = NULL;
+        glob_t files;
+
+        int res = glob(initrd_glob, 0, NULL, &files);
+        if (res == GLOB_NOMATCH) {
+                globfree(&files);
+                return true;
+        }
+        if (res != 0) {
+                globfree(&files);
+                return false;
+        }
+
+        for(size_t i = 0; i < files.gl_pathc; i++) {
+                initrd_source = files.gl_pathv[i];
+                initrd_target = string_printf("%s/%s", target_dir, basename(files.gl_pathv[i]));
```

Comment:
This is definitely going to leak. `autofree` uses the cleanup attribute and is scope based. As the variable declaration is *outside* this loop scope, it will only free in the unwinding of the function scope.

Easy remedy: Declare and initialise the autofree variable at the same time in this loop scope.

Reply:
Oof, I was clearly out of coffee. Updated!